pyflakes/checker.py

```diff
@@ -948,7 +1013,25 @@ def on_conditional_branch():
             except KeyError:
                 self.report(messages.UndefinedName, node, name)
 
+    def _handle_type_comments(self, node):
+        for (lineno, col_offset), comment in self._type_comments.get(node, ()):
+            comment = comment[len('# type:'):].strip()
+            func_match = re.match(r'^(\(.*?\)) *-> *(.*)$', comment)
```

Comment:
If I remember correctly, compiling the regular expression can be expensive. Should we do that once at the module level and re-use it over-and-over instead of compiling each time?

Reply:
you are correct, I thought I had hoisted these but looks like I forgot one!